readthedocs/core/permissions.py

```diff
@@ -1,18 +1,86 @@
-# -*- coding: utf-8 -*-
-
 """Objects for User permission checks."""
+from django.conf import settings
 from django.contrib.auth.models import User
 from django.db.models import Q
 
 from readthedocs.core.utils.extend import SettingsOverrideObject
+from readthedocs.organizations.constants import ADMIN_ACCESS, READ_ONLY_ACCESS
 
 
 class AdminPermissionBase:
 
     @classmethod
-    def admins(cls, obj):
+    def projects(cls, user, admin=False, member=False):
+        """
+        Return all the projects the user has access to as ``admin`` or ``member``.
+
+        If `RTD_ALLOW_ORGANIZATIONS` is enabled
+        this function takes SSO into consideration, it includes:
+
+        - Projects where the user has access to via SSO.
+        - Projects where the user has access via Team,
+          if SSO is not enabled for the organization of that team.
+
+        .. note::
+
+           SSO is taken into consideration,
+           but isn't implemented in .org yet.
+
+        :param user: user object to filter projects
+        :type user: django.contrib.admin.models.User
+        :param bool admin: include projects where the user has admin access to the project
+        :param bool member: include projects where the user has read access to the project
+        """
         from readthedocs.projects.models import Project
+        projects = Project.objects.none()
+        if not user or not user.is_authenticated:
+            return projects
+
+        if not settings.RTD_ALLOW_ORGANIZATIONS:
+            # All users are admin and member of a project
+            # when we aren't using organizations.
+            return user.projects.all()
+
+        if admin:
+            # Project Team Admin
+            admin_teams = user.teams.filter(access=ADMIN_ACCESS)
+            for team in admin_teams:
+                if not cls._has_sso_enabled(team.organization):
```

Comment:
It seems that you removed the `provider=` attribute from here. Why? (also in the other calls to `has_sso_enabled` from this method)

Reply:
We don't have the model that has those constants here, it was always using the email provider to check things, that is passed from .com.